packages/scm/src/browser/scm-tree-widget.tsx

```diff
@@ -65,13 +66,11 @@ export class ScmTreeWidget extends TreeWidget {
     ) {
         super(props, model, contextMenuRenderer);
         this.id = 'resource_widget';
-    }
-
-    @postConstruct()
-    protected init(): void {
-        super.init();
         this.addClass('groups-outer-container');
+    }
 
+    protected onAfterAttach(msg: Message): void {
+        super.onAfterAttach(msg);
         this.refreshOnRepositoryChange();
         this.toDispose.push(this.scmService.onDidChangeSelectedRepository(() => {
```

Comment:
`this.toDisposeOnDetach` otherwise it adds a new listener each time when a widget is moved between areas

Reply:
now changed to toDisposeOnDetach.